feat(op): comprehensive metrics documentation (#37924)
DataDog/documentation: content/en/observability_pipelines/sources/kafka.md

See the [librdkafka documentation][3] for more information and to ensure your values have the correct type and are within range.

## Metrics

For the [component metrics][10] and [source buffer metrics][13] emitted by all sources, see the Observability Pipelines Metrics documentation.

### Kafka metrics

- Use the `component_id` tag to filter or group by individual components.
- The `component_type` tag is `kafka` for these metrics.

`pipelines.kafka_consumer_lag`

> : **Description**: Kafka consumer lag per topic and partition. High values indicate the source is falling behind the incoming message rate.

Reviewer: Is it possible to give an example of what a hight value would be?

Author: as per the source lag, I lack knowledge to say what a high value would be, the trend is probably more meaningful than the values themselves.

: **Metric type**: gauge

`pipelines.kafka_consumed_messages_total`

> : **Description**: The number of messages consumed from Kafka brokers.

Reviewer: ```suggestion
: **Description**: The number of messages consumed by Kafka brokers.
```

Author: I think from is valid here and below. We are pulling events _from_ a kafka broker, not using a kafka broker to pull events

: **Metric type**: count

`pipelines.kafka_consumed_messages_bytes_total`

> : **Description**: The number of message bytes consumed from Kafka brokers.

Reviewer: ```suggestion
: **Description**: The number of message bytes the Worker receives from Kafka brokers.
```

: **Metric type**: count

`pipelines.kafka_requests_total`
: **Description**: The number of requests sent to Kafka brokers.
: **Metric type**: count

`pipelines.kafka_requests_bytes_total`
: **Description**: The number of bytes transmitted to Kafka brokers.
: **Metric type**: count

`pipelines.kafka_responses_total`

> : **Description**: The number of responses received from Kafka brokers.

Reviewer: ```suggestion
: **Description**: The number of responses received by Kafka brokers.
```

: **Metric type**: count

`pipelines.kafka_responses_bytes_total`

> : **Description**: The number of bytes received from Kafka brokers.

Reviewer: ```suggestion
: **Description**: The number of bytes received by Kafka brokers.
```
